File: apina-model/src/main/java/fi/evident/apina/model/parameters/EndpointPathVariableParameter.java
package fi.evident.apina.model.parameters;

import fi.evident.apina.model.type.ApiType;

import java.util.Optional;

public final class EndpointPathVariableParameter extends EndpointParameter {

    private final String pathVariable;

    public EndpointPathVariableParameter(String name, Optional<String> pathVariable, ApiType type) {
        super(name, type);

        this.pathVariable = pathVariable.orElse(name);
    }

    public String getPathVariable() {
        return pathVariable;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("@PathVariable");

        if (!pathVariable.equals(name))
            sb.append("(\"").append(pathVariable).append("\")");

        sb.append(' ').append(type).append(' ').append(name);
        return sb.toString();
    }
}

File: apina-core/src/main/kotlin/fi/evident/apina/model/parameters/EndpointPathVariableParameter.kt
package fi.evident.apina.model.parameters

import fi.evident.apina.model.type.ApiType

class EndpointPathVariableParameter(name: String, pathVariable: String?, type: ApiType) : EndpointParameter(name, type) {

    val pathVariable: String = pathVariable ?: name

    override fun toString(): String = buildString {
        append("@PathVariable")

        if (pathVariable != name)
            append("(\"").append(pathVariable).append("\")")

        append(' ').append(type).append(' ').append(name)
    }
}
